### PR TITLE
[platform_tests] add wait time for transceivers and processes

### PR DESCRIPTION
**Pull request by Xichen96**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [x] 202012
- [x] 202205
- [x] 202305
- [x] 202311

#### What is the motivation for this PR?
Wait time not enough for weak devices.

#### How did you do it?
Add wait time.

#### How did you verify/test it?
Run test

### CODE REVIEW

**Comment by lizhijianrd** [comment]
Can you please restrict the update to Celestica-E1031 only?

File: tests/platform_tests/test_reboot.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 ]
 
 
-MAX_WAIT_TIME_FOR_INTERFACES = 300
+MAX_WAIT_TIME_FOR_INTERFACES = 600
```
Comment (Blueve) [inline tests/platform_tests/test_reboot.py:31]: Please add a new wait time constant for E1031 only

File: tests/platform_tests/test_reboot.py [hunk 2]
```diff
@@ -39,9 +39,11 @@ def set_max_time_for_interfaces(duthost):
     to let MAX_TIME_TO_REBOOT to be overwritten by specified timeout value
     """
     global MAX_WAIT_TIME_FOR_INTERFACES
+    if duthost.facts["platform"] == "x86_64-cel_e1031-r0":
+        MAX_WAIT_TIME_FOR_INTERFACES = 600
     plt_reboot_ctrl = get_plt_reboot_ctrl(duthost, 'test_reboot.py', 'cold')
```
Comment (lizhijianrd) [inline tests/platform_tests/test_reboot.py:44]: Could you please check the comments of function [get_plt_reboot_ctrl](https://github.com/sonic-net/sonic-mgmt/blob/master/tests/common/utilities.py#L709)? Looks like we can define the timeout for E1031 in ansible inventory file.

File: tests/common/platform/processes_utils.py [hunk 3]
```diff
@@ -57,12 +57,13 @@ def check_critical_processes(dut, watch_secs=0):
         watch_secs = watch_secs - 5
 
 
-def wait_critical_processes(dut):
+def wait_critical_processes(dut, timeout=None):
```
Comment (lizhijianrd) [inline tests/common/platform/processes_utils.py:60]: Is it possible to identify the DUT platform from variable `dut`? If set the timeout to 600s for E1031 in this function, then it'll work everywhere.

**Review by lizhijianrd — APPROVED** [review]

**Review by wangxin — APPROVED** [review]

**Pull request merged**